File: app/src/main/java/network/devandroid/com/networklibrarycomparison/internal/NetFactory.java
package network.devandroid.com.networklibrarycomparison.internal;

import android.content.Context;

import network.devandroid.com.networklibrarycomparison.asyntask.AsyncNetworkManager;
import network.devandroid.com.networklibrarycomparison.fastnetworking.FastNetworkManager;
import network.devandroid.com.networklibrarycomparison.okhttp.OkHttpNetworkManager;
import network.devandroid.com.networklibrarycomparison.retrofit.RetrofitNetworkManager;
import network.devandroid.com.networklibrarycomparison.rxjava.RxJavaNetworkManager;
import network.devandroid.com.networklibrarycomparison.volley.VolleyNetworkManager;

public class NetFactory {

    public static BaseNetworkManager getNetworkManager(Context context, NetType type, INetworkManager.Callback callback) {
        switch (type) {
            case VOLLEY:
                return new VolleyNetworkManager(callback, context);
            case RETROFIT:
                return new RetrofitNetworkManager(callback);
            case OK_HTTP:
                return new OkHttpNetworkManager(callback);
            case FAST_NETWORK:
                return new FastNetworkManager(callback);
            case ASYNC_TASK:
                return new AsyncNetworkManager(callback);
            case RXJAVA:
                return new RxJavaNetworkManager(callback);
        }
        return null;
    }
}

File: app/src/main/java/network/devandroid/com/networklibrarycomparison/internal/NetFactory.kt
package network.devandroid.com.networklibrarycomparison.internal

import android.content.Context

import network.devandroid.com.networklibrarycomparison.asyntask.AsyncNetworkManager
import network.devandroid.com.networklibrarycomparison.fastnetworking.FastNetworkManager
import network.devandroid.com.networklibrarycomparison.okhttp.OkHttpNetworkManager
import network.devandroid.com.networklibrarycomparison.retrofit.RetrofitNetworkManager
import network.devandroid.com.networklibrarycomparison.rxjava.RxJavaNetworkManager
import network.devandroid.com.networklibrarycomparison.volley.VolleyNetworkManager

object NetFactory {

    fun getNetworkManager(context: Context, type: NetType, callback: INetworkManager.Callback): BaseNetworkManager? {
        when (type) {
            NetType.VOLLEY -> return VolleyNetworkManager(callback, context)
            NetType.RETROFIT -> return RetrofitNetworkManager(callback)
            NetType.OK_HTTP -> return OkHttpNetworkManager(callback)
            NetType.FAST_NETWORK -> return FastNetworkManager(callback)
            NetType.ASYNC_TASK -> return AsyncNetworkManager(callback)
            NetType.RXJAVA -> return RxJavaNetworkManager(callback)
        }
        return null
    }
}
